src/lib/battery/battery.cpp

```diff
@@ -207,18 +205,23 @@ void Battery::estimateStateOfCharge(const float voltage_v, const float current_a
 	if (sees_coulomb_counting_only) {
 
 		// This ensures the cell voltage is derived from the filtered voltage and corrects for any voltage load drop as PX4 uses the raw voltage values.
-
 		cell_voltage_filtered = _voltage_filter_v.getState();
 		cell_voltage_filtered /= _params.n_cells;
 		cell_voltage_filtered_load += _params.r_internal * current_a;
 
-		// Derives initial SoC Estimation from the voltage. A 15 second timer is included to allow voltage to settle whilst components initialise.
-		if(first_run || ((hrt_absolute_time() - first_run_time) < 15'000'000)){
-			struct Lookup {
-				float OCVoltage;
-				float SOC;
-			};
-				Lookup SOCLookup[] {
+		// Initially dervied an initial SoC Estimation from the voltage at low current draw. Tiree in idle state draws ~3A and Birdham ~0.5A, so idle threshold is set to 4A.
+		// However, it's difficult to characterise idle current in a way that is appropriate for both Birdham and Tiree.
+		// Must be disarmed to prevent any sporadic drops in current from triggering voltage-based estimation midflight (battery voltage under load is unrepresentative of SoC).
+		// Also eliminates risk of takeoff affecting initial SoC.
+		if (_vehicle_control_mode_sub.updated()) {
+			vehicle_control_mode_s vehicle_control_mode;
+			if (_vehicle_control_mode_sub.copy(&vehicle_control_mode)) {
+				_armed = vehicle_control_mode.flag_armed;
+			}
+		}
+
+		if(!_armed){
```

Comment:
Moved Lookup struct definition to header.


Voltage-based SoC estimation was first done for 15s at startup, however this isn't a particularly versatile approach.

Instead, a 'voltage-based at low current draw' approach was developed but it's difficult to characterise a generic 'idle' current across different drones (Tiree, Birdham, Yealm).
Also, if this 'idle' current varies across flight days, then this fix becomes less effective.
E.g initial SoC estimation can be affected by voltage drop at takeoff (before defined 'idle' current is exceeded) where even a 0.05 drop in voltage can cause between 5-20% drop in SoC.

Altered estimation to be derived from voltage when !armed instead of at low current.